src/main/java/com/internet/shop/controller/DeleteProductFromShoppingCartController.java

```diff
@@ -0,0 +1,31 @@
+package com.internet.shop.controller;
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.service.ShoppingCartService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class DeleteProductFromShoppingCartController extends HttpServlet {
+    private static final Long USER_ID = 1L;
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private ShoppingCartService cartService = (ShoppingCartService) injector
+            .getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        ShoppingCart shoppingCart = cartService.getByUserId(USER_ID);
+        long id = Long.parseLong(req.getParameter("id"));
```

Comment:
can we just get product by id from storage the same way we did with shoppingCart and then call cartService.deleteProduct(shoppingCart, product) ?

Reply:
done